File: app/src/main/java/com.example.safehome/Door/DoubleDoorFactory.java
package com.example.safehome.Door;

import com.example.safehome.Door.DoorFactory;

public class DoubleDoorFactory implements DoorFactory<Door> {
    /**
     * FactoryMethod returns class of type Door.
     *
     * @param request
     * @return FactoryMethod returns class of type Door.
     */
    @Override
    public Door factoryMethod(Door request) {
        return null;
    }
}

File: app/src/main/kotlin/com.example.safehome/Door/DoubleDoorFactory.kt
package com.example.safehome.Door;

import org.slf4j.Logger;
import org.slf4j.LoggerFactory;


public class DoubleDoorFactory implements DoorFactory<Door> {
    private static final Logger log = LoggerFactory.getLogger(DoubleDoorFactory.class);
    /**
     * FactoryMethod returns class of type Door.
     *
     * @param request
     * @return FactoryMethod returns class of type Door.
     */
    @Override
    public Door factoryMethod(Door request) {
        return null;
    }
}